File: dmoj/boxes.cpp
#include <bits/stdc++.h>
using namespace std;
#pragma GCC optimize "Ofast"

bool compare(const vector<int>& a, const vector<int>& b) {
    int am = 1;
    int bm = 1;
    for(const auto&num:a) {
        am *= num;
    }
    for(const auto&num:b) {
        bm *= num;
    }
    return(am < bm);
}

int main() {
    int N;
    scanf("%i", &N);
    vector< vector<int> > boxes(N);
    for(int i = 0; i < N; i++) {
        for(int j = 0; j < 3; j++) {
            int dimension;
            scanf("%i", &dimension);
            boxes[i].push_back(dimension);
        }
    }
    sort(boxes.begin(), boxes.end(), compare);

    int M;
    scanf("%i", &M);
    int toBPackaged [M][3];
    for(int i = 0; i < M; i++) {
        for(int j = 0; j < 3; j++) {
            int dim;
            scanf("%i", &dim);
            toBPackaged[i][j] = dim;
        }
    }

    for(auto package: toBPackaged) {
        bool foundBoxThatFits = false;
        long int boxSize;
        for(auto box: boxes) {
            int w = package[0]; int l = package[1]; int h = package[2];
            if(
            (w <= box[0] && l <= box[1] && h <= box[2]) || 
            (w <= box[0] && h <= box[1] && w <= box[2]) || 

            (h <= box[0] && l <= box[1] && w <= box[2]) ||  
            (h <= box[0] && w <= box[1] && l <= box[2]) ||

            (l <= box[0] && h <= box[1] && w <= box[2]) || 
            (l <= box[0] && w <= box[1] && h <= box[2]) 
            ) {
                foundBoxThatFits = true;
                boxSize = box[0] * box[1] * box[2];
                break;
            }

        }
        if(foundBoxThatFits) {
            printf("%li\n", boxSize);
        } else {
            printf("Item does not fit.\n");
        }
    }
}

File: dmoj/Boxes.java
import java.util.*;

public class Boxes {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[][] boxes = new int[N][3];
        
        for(int r = 0; r < N; r++) {
            boxes[r][0] = sc.nextInt();
            boxes[r][1] = sc.nextInt();
            boxes[r][2] = sc.nextInt();
            Arrays.sort(boxes[r]);
        }

        int M = sc.nextInt();
        for(int i = 0; i < M; i++) {
            int minV = Integer.MAX_VALUE;
            int[] item = new int[3];
            for(int j = 0; j < 3; j++)
                item[j] = sc.nextInt();
            
            Arrays.sort(item);

            for(int r = 0; r < N; r ++)
                if(item[0] <= boxes[r][0] && item[1] <= boxes[r][1] && item[2] <= boxes[r][2]) {
                    int vol = boxes[r][0] * boxes[r][1] * boxes[r][2];
                    if(vol < minV)
                        minV = vol;
                }

            if(minV == Integer.MAX_VALUE)
                System.out.println("Item does not fit.");
            else
                System.out.println(minV);
        }

        sc.close();
    }
}
